CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2213,7 +2213,7 @@ protected function processFormSubmission($contactID) {
           }
         }
       }
-      $paymentParams['amount'] = $this->getMainContributionAmount();
+      $paymentParams['amount'] = $this->_params['amount'];
```

Comment:
This seems like the place where you should do `$this->getPartialPaymentAmount() ?: $this->getMainContributionAmount()`

& then $this->getPartialPaymentAmount() does something like

``
if ($isPartialPayment) {
  return `$this->getSubmittedValue('total_amount');`
}
return NULL
``